src/toil_scripts/gatk_germline/vqsr.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python2.7
+from __future__ import print_function
+import multiprocessing
+import os
+
+from toil.job import PromisedRequirement
+from bd2k.util.humanize import human2bytes
+
+from toil_scripts.tools.variant_filters import gatk_genotype_gvcf, \
+    gatk_variant_recalibrator, gatk_apply_variant_recalibration
+from toil_scripts.lib.files import upload_or_move_job
+
+
+def vqsr_pipeline(job, gvcfs, config):
+    """
+    Runs GATK Variant Quality Score Recalibration. Writes VQSR VCF to an output directory defined
+    in the config dictionary. Multiple GVCFs are joined and recalibrated into a single VCF.
+
+    0: GenotypeGVCFs                0 --> 1 --> 3 --> 4 --> 5
+    1: Recalibrate SNPs                   |      |
+    2: Recalibrate INDELS                 +-> 2 -+
+    3: Apply SNPs
+    4: Apply INDELS
+    5: Write VCF to output directory
+
+    :param dict gvcfs: Dictionary of UUIDs and GVCF FileStoreIDs
+    :param Namespace config: Input parameters
+    :return: SNP and INDEL VQSR VCF FileStoreID
+    :rtype: str
+    """
+    cores = multiprocessing.cpu_count()
+    genotype_gvcf_disk = PromisedRequirement(
+        lambda lst: 2*sum(x.size for x in lst)+human2bytes('5G'), gvcfs.values())
+    genotype_gvcf = job.wrapJobFn(gatk_genotype_gvcf,
+                                  gvcfs,
+                                  config,
+                                  disk=genotype_gvcf_disk, cores=cores)
+
+    snp_recal_disk = PromisedRequirement(lambda x: 2*x.size+human2bytes('10G'), genotype_gvcf.rv())
+    snp_recal = job.wrapJobFn(gatk_variant_recalibrator,
+                              'SNP',
+                              genotype_gvcf.rv(),
+                              config,
+                              disk=snp_recal_disk, cores=cores/2)
+
+    indel_recal_disk = PromisedRequirement(lambda x: 2*x.size+human2bytes('10G'), genotype_gvcf.rv())
+    indel_recal = job.wrapJobFn(gatk_variant_recalibrator,
+                                'INDEL',
+                                genotype_gvcf.rv(),
+                                config,
+                                disk=indel_recal_disk, cores=cores/2)
+
+    apply_snp_recal_disk = PromisedRequirement(
+        lambda x,y,z: 2*(x.size + y.size + z.size) + human2bytes('5G'),
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.